src/ir/function.rs

```diff
@@ -309,6 +309,31 @@ pub fn cursor_mangling(
     Some(mangling)
 }
 
+fn args_from_ty_and_cursor(
+    ty: &clang::Type,
+    cursor: &clang::Cursor,
+    ctx: &mut BindgenContext,
+) -> Vec<(Option<String>, TypeId)> {
```

Comment:
So the key is going through `ty.args()` first if possible if this is a function prototype, right? That makes perfect sense to me!

Reply:
Yep!